### PR TITLE
Add simple scripts for checking if generated code changed.

### PR DESCRIPTION
**Pull request by gchanan**
This is designed to make it easier to see how your codegen changes affected actual generated code.

Limitations:
A) This is NOT robust; if new directories are added that include generated files, they need to be added to tools/generated_dirs.txt.  Note that subdirectories of the list are not included.

B) This is particular to my workflow which I don't claim is generally applicable.  Ideally we would have a script that pumped out a diff that could be attached to PRs.

C) Only works on OSS and definitely won't work on windows.

How to use:
1) python setup.py ...
2) tools/git_add_generated_dirs
3) Edit codegen
4) python setup.py ...
4) git diff to see changes
5) If satisfied: tools/git_reset_generated_dirs, commit, etc.
   If not satisfied: Go to 3)

### CODE REVIEW

**Comment by gchanan** [comment]
I am happy to be told these are too specific for my workflow and we shouldn't bless it by putting this in the tree.  My real hope is that someone takes this and makes it actually generally useful.

File: tools/git_add_generated_dirs.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+BASEDIR=$(dirname "$0")
+(< $BASEDIR/generated_dirs.txt xargs -i find {} -type f) | xargs git add -f
```
Comment (ezyang) [inline tools/git_add_generated_dirs.sh]: woooow I didn't know you could put the `<` before the command. Fancy shmancy.

For max portability, you should put a shebang on the shell script

**Review by ezyang — CHANGES_REQUESTED** [review]
The scripts themselves should have the docs you wrote in the commit.

File: tools/git_reset_generated_dirs.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+# HOW TO USE:
```
Comment (ezyang) [inline tools/git_reset_generated_dirs.sh]: You could use a `Note [git generated dirs]` script to avoid having to copy paste it two places ;)

**Review by ezyang — APPROVED** [review]

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**